tests/unit-tests/CMakeLists.txt

```
@@ -409,6 +409,15 @@ target_link_libraries(
   -lpthread -lm
 )
 
+target_compile_definitions(
+        compute_tests
+        PUBLIC
+        $<$<BOOL:${__APPLE__}>:-DSPECFEMPP_APPLE_BUILD>
+        $<$<BOOL:${__GNU__}>:-DSPECFEMPP_GNU_BUILD>
+        $<$<BOOL:${__MSVC__}>:-DSPECFEMPP_MSVC_BUILD>
+        $<$<BOOL:${__INTEL_LLVM__}>:-DSPECFEMPP_INTEL_LLVM_BUILD>
```

Also, `-D` only needed for `target_compile_options`

```suggestion
        $<$<BOOL:${__APPLE__}>:SPECFEMPP_APPLE_BUILD>
        $<$<BOOL:${__GNU__}>:SPECFEMPP_GNU_BUILD>
        $<$<BOOL:${__MSVC__}>:SPECFEMPP_MSVC_BUILD>
        $<$<BOOL:${__INTEL_LLVM__}>:SPECFEMPP_INTEL_LLVM_BUILD>
```